codec/conllu.py

```diff
@@ -124,10 +129,10 @@ def compute(form):
   return graph;
 
 def read(stream, framework = None, text = None, anchors = None, trace = 0):
-  id, input, tuples = read_tuples(stream);
-  while tuples:
+  tuples_generator = read_tuples(stream)
+  for id, input, tuples in tuples_generator:
     if trace:
       print("conllu.read(): processing graph #{} ...".format(id),
             file = sys.stderr);
-    yield construct_graph(id, input, tuples, framework, text, anchors), None;
-    id, input, tuples = read_tuples(stream);
+    if tuples:
```

Comment:
I think the "hack" of returning `tuples` as `None` (or `[]` for that matter) is no longer necessary when this is a generator, since you can just yield only the actual `id, input, tuples` triples. Do you agree? Would you simplify it? Thanks!